dds/DCPS/Recorder.h

```diff
@@ -12,10 +12,11 @@
 #include "LocalObject.h"
 #include "RcHandle_T.h"
 #include "RawDataSample.h"
-#include "dds/DdsDcpsInfrastructureC.h"
+#include "GuidConverter.h"
 
-#include "XTypes/TypeObject.h"
-#include "XTypes/DynamicType.h"
+#include "XTypes/DynamicData.h"
+
+#include <dds/DdsDcpsInfrastructureC.h>
```

Comment:
I guess something in the `#include`s is causing the error in build_u18_d0i0v1_sec_js0_FM-08, but I don't know what the root cause is.

Try changing line 125 to just use `bool`.  There is no need to use a typedef for `bool` when the type is always `bool`.  Maybe there's some other problem in CI.

Also try making the minimal change that you need in these `#includes` in the header, moving other `#include`s to the cpp.